plugins/package-managers/pub/src/main/kotlin/Pub.kt

```diff
@@ -105,7 +105,7 @@ internal class PubCommand(private val flutterAbsolutePath: File) : CommandLineTo
     override fun transformVersion(output: String) = output.removePrefix("Dart SDK version: ").substringBefore(' ')
 
     override fun command(workingDir: File?): String =
-        if (flutterAbsolutePath.isDirectory) "$flutterAbsolutePath${File.separator}$dartCommand" else dartCommand
+        if (flutterAbsolutePath.isAbsolute) "$flutterAbsolutePath${File.separator}$dartCommand" else dartCommand
```

Comment:
I believe passing `File("")` to the constructor is wrong, because it is not an absolute path.
Understanding this commit to me was not a super straight forward experience for me, due to the    mis-leading use of `flutterAbsolutePath`. 

Instead of this commit I propose to make
1. the `flutterAbsolutePath` nullable
2. the primary constructor have a require check to ensure the given `flutterAbsolutePath` actually is absolute and exists (ìsDirectory).
3. have the case distinction in `command()` only based on whether the `flutterAbsolutePath` is null.